src/sentry/eventstore/models.py

```diff
@@ -476,9 +456,6 @@ def get_grouping_variants(
     def get_primary_hash(self) -> str | None:
```

Comment:
Another follow-up thing to think about: Can this ever actually return `None`? It would save a lot of noise in tests if we didn't have to assert its existence all the time.

Reply:
The original code was never None:
https://github.com/getsentry/sentry/blob/492e78e0acb7fa508c53ffeea5e9c622db8202a5/src/sentry/eventstore/models.py#L385-L387

Reply:
Ah, is that from before hierarchical grouping? In that case, yeah, not in this set of PRs, but it'll be great to pull that `None` from the return type and get rid of all of the non-None assertions in tests.